crates/environ/src/fact/trampoline.rs

```diff
@@ -404,16 +817,28 @@ impl Compiler<'_, '_> {
         let dst = if let Some(flat) = &dst_flat {
             Destination::Stack(flat, lift_opts)
         } else {
-            // If there are too many parameters then space is allocated in the
-            // destination module for the parameters via its `realloc` function.
-            let abi = CanonicalAbiInfo::record(dst_tys.iter().map(|t| self.types.canonical_abi(t)));
-            let (size, align) = if lift_opts.memory64 {
-                (abi.size64, abi.align64)
+            if lift_opts.async_ {
+                let align = dst_tys
+                    .iter()
+                    .map(|t| self.types.align(lift_opts, t))
+                    .max()
+                    .unwrap_or(1);
```

Comment:
How come the `::record` constructor couldn't be used like before?

Reply:
Ah, I think this is an artifact of when Luke changed the async lift ABI; previously parameter passing worked differently for async vs. sync lifts, and now they're the same.  In the churn I neglected to unify the code.  I'll do that.